utils.h

```diff
@@ -0,0 +1,65 @@
+#ifndef _UTILS_H
+#define _UTILS_H
+
+#include <stdio.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "php_mysqlnd_azure.h"
+
+extern FILE *logfile;
+#define TIME_FORMAT "%Y-%m-%d %H:%M:%S"
+
+// Azure Log Levels, 1(ERROR), 2(INFO), 3(DEBUG)
+#define ALOGERR  1
+#define ALOGINFO 2
+#define ALOGDBG  3
+
+#define OPEN_LOGFILE(filename)                                                \
+  do {                                                                        \
+    if (filename != NULL) {                                                   \
+      logfile = fopen(filename, "a");                                         \
```

Comment:
fopen will fail, which logfile will be null, take care of this when we write logs, I see you already checked this in close.